pkg/apis/paddlepaddle/v1/types.go

```diff
@@ -0,0 +1,171 @@
+	// If you want to use the hostnetwork instead of container network
+	// portmanager is necessary. About portmanager, please refer to
+	// https://github.com/PaddlePaddle/cloud/blob/develop/doc/hostnetwork/hostnetwork.md
+	HostNetwork       bool                 `json:"host_network,omitempty"`
+	Port              int                  `json:"port,omitempty"`
+	PortsNum          int                  `json:"ports_num,omitempty"`
+	PortsNumForSparse int                  `json:"ports_num_for_sparse,omitempty"`
+	FaultTolerant     bool                 `json:"fault_tolerant,omitempty"`
+	Passes            int                  `json:"passes,omitempty"`
+	Volumes           []corev1.Volume      `json:"volumes"`
+	VolumeMounts      []corev1.VolumeMount `json:"VolumeMounts"`
+	//TrainingJob components.
+	Master  MasterSpec  `json:"master"`
+	Pserver PserverSpec `json:"pserver"`
+	Trainer TrainerSpec `json:"trainer"`
+}
+
+// MasterSpec is the spec for a master in the paddle job
+type MasterSpec struct {
+	EtcdEndpoint string                      `json:"etcd-endpoint"`
+	Resources    corev1.ResourceRequirements `json:"resources"`
+	ReplicaSpec  *v1beta1.ReplicaSet         `json:"replicaSpec"`
+}
+
+// PserverSpec is the spec for pservers in the paddle job
+type PserverSpec struct {
+	MinInstance int                         `json:"min-instance"`
+	MaxInstance int                         `json:"max-instance"`
+	Resources   corev1.ResourceRequirements `json:"resources"`
+	ReplicaSpec *v1beta1.ReplicaSet         `json:"replicaSpec"`
+}
+
+// TrainerSpec is the spec for trainers in the paddle job
+type TrainerSpec struct {
+	EtcdEndpoint string                      `json:"etcd-endpoint"`
+	Entrypoint   string                      `json:"entrypoint"`
+	Workspace    string                      `json:"workspace"`
+	MinInstance  int                         `json:"min-instance"`
+	MaxInstance  int                         `json:"max-instance"`
+	Resources    corev1.ResourceRequirements `json:"resources"`
+	ReplicaSpec  *batchv1.Job                `json:"replicaSpec"`
+}
+
+// TrainingJobPhase is the phase of TrainingJob
+type TrainingJobPhase string
+
+const (
+	// TrainingJobPhaseNone is empty TrainingJobPhase.
+	TrainingJobPhaseNone TrainingJobPhase = ""
+	// TrainingJobPhaseCreating is creating TrainingJobPhase.
+	TrainingJobPhaseCreating = "creating"
+	// TrainingJobPhaseRunning is running TrainingJobPhase.
+	TrainingJobPhaseRunning = "running"
+	// TrainingJobPhaseSucceeded is succeeded TrainingJobPhase.
+	TrainingJobPhaseSucceeded = "succeeded"
+	// TrainingJobPhaseFailed is failed TrainingJobPhase.
+	TrainingJobPhaseFailed = "failed"
+)
+
+// TrainerJobScaleStatus is status of trainer jobs.
+type TrainerJobScaleStatus struct {
+}
+
+// TrainingResourceType the type of TrainingJob resource, include MASTER PSERVER and TRAINER
+type TrainingResourceType string
+
+const (
+	// MASTER is the master name of TrainingResourceType.
+	MASTER TrainingResourceType = "MASTER"
```

Comment:
Is it a Kubernetes convention to have it named all capital case (MASTER vs Master). The Go convention is camel case (Master).

Reply:
I'll change the format in the next pr.